packages/json-schema/test/integration/build-schema.test.ts

```diff
@@ -0,0 +1,216 @@
+    });
+    it('does not convert models that have not been decorated with @model()', () => {
+      class Empty {}
+      class NoModelMeta {
+        @property() foo: string;
+        bar: number;
+      }
+      expect(modelToJsonDef(Empty)).to.eql({});
+      expect(modelToJsonDef(NoModelMeta)).to.eql({});
+    });
+    it('properly converts string, number, and boolean properties', () => {
+      @model()
+      class TestModel {
+        @property() str: string;
+        @property() num: number;
+        @property() bool: boolean;
+      }
+
+      const jsonDef = modelToJsonDef(TestModel);
+      expect(jsonDef.properties).to.deepEqual({
+        str: {
+          type: 'string',
+        },
+        num: {
+          type: 'number',
+        },
+        bool: {
+          type: 'boolean',
+        },
+      });
+    });
+    it('properly converts object properties', () => {
+      @model()
+      class TestModel {
+        @property() obj: object;
+      }
+
+      const jsonDef = modelToJsonDef(TestModel);
+      expect(jsonDef.properties).to.deepEqual({
+        obj: {
+          type: 'object',
+        },
+      });
+    });
+    it('properly converts custom type properties', () => {
+      class CustomType {
+        prop: string;
+      }
+
+      @model()
+      class TestModel {
+        @property() cusType: CustomType;
+      }
+
+      const jsonDef = modelToJsonDef(TestModel);
+      expect(jsonDef.properties).to.deepEqual({
+        cusType: {
+          $ref: '#definitions/CustomType',
+        },
+      });
+    });
+    it('properly converts primitive arrays properties', () => {
+      @model()
+      class TestModel {
+        @property.array(Number) numArr: number[];
+      }
+
+      const jsonDef = modelToJsonDef(TestModel);
+      expect(jsonDef.properties).to.deepEqual({
+        numArr: {
+          type: 'array',
+          items: {
+            type: 'number',
+          },
+        },
+      });
+    });
+    it('properly converts custom type arrays properties', () => {
+      class CustomType {
+        prop: string;
+      }
+
+      @model()
+      class TestModel {
+        @property.array(CustomType) cusArr: CustomType[];
+      }
+
+      const jsonDef = modelToJsonDef(TestModel);
+      expect(jsonDef.properties).to.deepEqual({
+        cusArr: {
+          type: 'array',
+          items: {
+            $ref: '#definitions/CustomType',
+          },
+        },
+      });
+    });
+    describe('toJSONProperty', () => {
+      class Bar {
+        barA: number;
+      }
+      @model()
+      class Foo {
+        @property() str: string;
+        @property() num: number;
+        @property() bool: boolean;
+        @property() obj: object;
+        @property() nul: null;
+        @property() undef: undefined;
+        @property.array(String) arrStr: string[];
+        @property() bar: Bar;
+        @property.array(Bar) arrBar: Bar[];
+      }
+      // tslint:disable-next-line:no-any
+      let meta: any;
```

Comment:
I think `meta` should have type `ModelDefinitionSyntax`, shouldn't it? See [here](https://github.com/strongloop/loopback-next/blob/a50405a5a56b5f1e509a2c30d984fd262a342cba/packages/repository/src/decorators/model.ts#L31).

I think this is showing a possibly missing return-type-signature in `getModelMetadata` method (see [here](https://github.com/strongloop/loopback-next/blob/a50405a5a56b5f1e509a2c30d984fd262a342cba/packages/repository/src/decorators/metadata.ts#L12)). 

Please improve the existing code in other packages in such way that this test does not need to use `any` and can use a named type instead.